src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/W3CPropagator.cs

```diff
@@ -111,7 +111,9 @@ internal static bool TryExtractBaggage(string? baggageString, out IEnumerable<Ke
                 if (TryDecodeBaggageKey(keySpan, out string? key) && TryDecodeBaggageValue(valueSpan, out string value))
                 {
                     baggageList ??= new List<KeyValuePair<string, string?>>();
-                    baggageList.Add(new KeyValuePair<string, string?>(key, value));
+
+                    // Insert in reverse order for asp.net compatibility.
+                    baggageList.Insert(0, new KeyValuePair<string, string?>(key, value));
```

Comment:
is there a reason you didn't use baggageList.Reverse() affter the Add calls instead?

Reply:
We can do that. Thanks!